=== modified file 'account_journal/account_journal.py'
--- account_journal/account_journal.py	2011-09-08 13:49:03 +0000
+++ account_journal/account_journal.py	2011-09-08 08:28:32 +0000
@@ -74,7 +74,8 @@
         return res
     
     def create(self, cr, uid, vals, context=None):
-        
+        if not context:
+            context = {}
         # TODO: add default accounts
         
         # Create associated sequence
@@ -107,7 +108,7 @@
         journal_obj = super(account_journal, self).create(cr, uid, vals, context)
         
         # if the journal can be linked to a register, the register is also created
-        if vals['type'] in ('cash','bank','cheque'):
+        if vals['type'] in ('cash','bank','cheque') and not context.get('sync_data'):
             self.pool.get('account.bank.statement') \
                 .create(cr, uid, {'journal_id': journal_obj,
                                   'name': "REG1" + vals['code'],

